File: Array/2073-TimeNeededToBuyTickets .py
def timeRequiredToBuy(tickets, k):
    sec = 0

    while tickets[k] != 0:
        for t, i in tickets:
            if tickets[i] > 0:
                tickets[i] -= 1
                sec += 1

            if tickets[k] == 0:
                return sec

    return sec


def timeRequiredToBuy2(tickets, k):
    sec = 0

    for i in range(len(tickets)):
        if i <= k:
           sec += min(tickets[i], tickets[k])
        else:
            sec += min(tickets[i], tickets[k]-1)

    return sec

File: 2073_TimeNeededToBuyTickets.ts
function timeRequiredToBuy2(tickets: number[], k: number): number {
	let sec = 0

	while (tickets[k] > 0) {
		for (let i = 0; i < tickets.length; i++) {
			if (tickets[i] !== 0) {
				tickets[i] -= 1
				sec++
			}

			if (tickets[k] === 0) break
		}
	}

	return sec
}

function timeRequiredToBuy(tickets: number[], k: number): number {
	let sec = 0

	for (let i = 0; i < tickets.length; i++) {
		if (i <= k) {
			sec += Math.min(tickets[i], tickets[k])
		} else {
			sec += Math.min(tickets[i], tickets[k] - 1)
		}
	}

	return sec
}

console.log(timeRequiredToBuy([5, 1, 1, 1], 0))
console.log(timeRequiredToBuy([84, 49, 5, 24, 70, 77, 87, 8], 3))
